wallet-new/src/Cardano/Wallet/API/WIP/LegacyHandlers.hs

```diff
@@ -20,51 +27,140 @@ import qualified Pos.Wallet.Web.State as V0 (WalletSnapshot, askWalletSnapshot)
+        txInputs  <- toPaymentDistributions ctInputs
+        txOutputs <- toPaymentDistributions ctOutputs
+
+        let txType = if ctIsLocal
+            then V1.LocalTransaction
+            else V1.ForeignTransaction
+
+        let txDirection = if ctIsOutgoing
+            then V1.OutgoingTransaction
+            else V1.IncomingTransaction
+
+        let V0.CTxMeta{..} = ctMeta
+        txCreationTime <- toTimestamp ctmDate
+        txStatus <- toTransactionStatus ctCondition
+
+        pure V1.Transaction{..}
+
+toTransaction :: Catch.MonadThrow m  => V0.CTx -> m Transaction
+toTransaction from = case toTransactionEither from of
+    Left e   -> Catch.throwM e
+    Right to -> pure to
+
+toTxId :: V0.CTxId -> Either MigrationError (V1 Txp.TxId)
+toTxId (V0.CTxId (V0.CHash h)) =
+        let err = Left . MigrationFailed . mappend "Error migrating a TxId: "
+        in either err (pure . V1) (decodeHash h)
+
+toCoin :: V0.CCoin -> Either MigrationError (V1 Core.Coin)
+toCoin c =
+    let err = Left . MigrationFailed . mappend "error migrating V0.CCoin -> Core.Coin, mkCoin failed: "
+    in either err (pure . V1) (decodeCType c)
+
+toAddress :: V0.CId V0.Addr -> Either MigrationError (V1 Core.Address)
+toAddress (V0.CId (V0.CHash h)) =
+    let err = Left . MigrationFailed . mappend "Error migrating (V0.CId V0.Addr) -> Core.Address failed."
+    in either err (pure . V1) (Core.decodeTextAddress h)
+
+toPaymentDistribution :: (V0.CId V0.Addr, V0.CCoin) -> Either MigrationError V1.PaymentDistribution
+toPaymentDistribution (cIdAddr, cCoin) = do
+    pdAddress <- toAddress cIdAddr
+    pdAmount  <- toCoin cCoin
+    pure V1.PaymentDistribution {..}
+
+toPaymentDistributions :: [(V0.CId V0.Addr, V0.CCoin)] -> Either MigrationError (NonEmpty V1.PaymentDistribution)
+toPaymentDistributions [] = Left $ MigrationFailed "Failed migrating because the list was empty."
+toPaymentDistributions (x:xs) =
+    (:|) <$> toPaymentDistribution x <*> mapM toPaymentDistribution xs
+
+toTimestamp :: POSIXTime -> Either MigrationError (V1 Core.Timestamp)
+toTimestamp time =
+    pure . V1 $ view (Lens.from Core.timestampSeconds) time
+
+toTransactionStatus :: V0.CPtxCondition -> Either MigrationError (V1.TransactionStatus)
+toTransactionStatus = Right . \case
+    V0.CPtxCreating{} ->
+        V1.Creating
+    V0.CPtxApplying{} ->
+        V1.Applying
+    V0.CPtxInBlocks{} ->
+        V1.InNewestBlocks
+    V0.CPtxWontApply{} ->
+        V1.WontApply
+    V0.CPtxNotTracked ->
+        V1.Persisted
+
+fromAssuranceLevelEither :: V1.AssuranceLevel -> Either MigrationError V0.CWalletAssurance
+fromAssuranceLevelEither V1.StrictAssurance = pure V0.CWAStrict
+fromAssuranceLevelEither V1.NormalAssurance = pure V0.CWANormal
+
+fromAssuranceLevel :: Catch.MonadThrow m  => V1.AssuranceLevel -> m V0.CWalletAssurance
+fromAssuranceLevel from = case fromAssuranceLevelEither from of
+    Left e   -> Catch.throwM e
+    Right to -> pure to
+
```

Comment:
Minor, we can change on the story-branch directly when merging but `MonadThrow` is part of `Universum` so there's no need for the import above, and even less need for a qualified one ;) ... 
As a rule of thumb, prefer explicit imports over qualified ones. Used qualified imports when functions are ambiguous / context-related. For instance: `fromList`, `encode`, `toString` ... In such case, we'd prefer:

`Map.fromList`, `Yaml.encode` and `Scientific.toString` for example.

Reply:
Right!!! I can make this change in PR concerning CO-383 :-)